kitsune/questions/views.py

```diff
@@ -636,8 +636,9 @@ def aaq_step3(request, product_key, category_key=None):
     # Since removing the @login_required decorator for MA form
     # need to catch unauthenticated, non-MA users here """
     is_loginless = product_key in settings.LOGIN_EXCEPTIONS
+    skip_auth = request.GET.get("skip_auth", False)
 
-    if not is_loginless and not request.user.is_authenticated:
+    if not request.user.is_authenticated and not skip_auth:
```

Comment:
As it stands, this is a loophole that would allow users to skip login for any product key, just by adding `?skip_auth=True` to `/{locale}/questions/new/{product-key}/form`, although the flow after skipping the login will probably raise an exception at some point since it'll expect a non-anonymous user with attributes like `email` defined. In any case, we should only respect `skip_auth` when `is_loginless` is true.

```suggestion
    if not ((is_loginless and skip_auth) or request.user.is_authenticated):
```

Reply:
Thanks!